index.js

```diff
@@ -219,16 +219,21 @@ HyperDB.prototype.heads = function (cb) {
   if (!this.opened) return readyAndHeads(this, cb)
   if (this._heads) return process.nextTick(cb, null, this._heads)
 
+  var self = this
+
   // This is a bit of a hack. Basically when the db is empty
   // we wanna wait for data to come in. TODO: We should guarantee
   // that the db always has a single block of data (like a header)
   if (this._waitForUpdate()) {
     this.setMaxListeners(0)
-    this.once('remote-update', this.heads.bind(this, cb))
+    this.once('remote-update', function () {
+      process.nextTick(function () {
```

Comment:
I added this nextTick() because hypercore doesn't update the length until after the remote-update event is sent.

Reply:
which length are you talking about here? https://github.com/mafintosh/hypercore/blob/master/lib/replicate.js#L212 seems to trigger after remoteLength is updated above.